c2rust-transpile/src/c_ast/conversion.rs

```diff
@@ -1261,7 +1261,11 @@ impl ConversionContext {
                 }
 
                 ASTEntryTag::TagStringLiteral if expected_ty & (EXPR | STMT) != 0 => {
-                    let ty_old = node.type_id.expect("Expected expression to have type");
+                    // https://github.com/immunant/c2rust/issues/1332
+                    // string literals from assert message like this expression
+                    // _Static_assert(1, "message")
+                    //  don't have type. so we give it zero for now
```

Comment:
```suggestion
                    // See <https://github.com/immunant/c2rust/issues/1332>.
                    // String literals from `_Static_assert`s don't have a type,
                    // so we give it 0 for now because ...
```

Something like this with more of an explanation for what 0 means would be helpful.